src/merkle/mmr/accumulator.rs

```diff
@@ -59,3 +59,23 @@ impl MmrPeaks {
             .verify(opening.relative_pos() as u64, value, root)
     }
 }
+
+impl From<&MmrPeaks> for Vec<Felt> {
+    fn from(mmr: &MmrPeaks) -> Self {
+        // Account for padding
+        let num_peaks = mmr.peaks.len();
+        let len = if num_peaks < 16 {
+            65
+        } else if num_peaks % 2 == 1 {
+            (num_peaks + 1) * 4 + 1
+        } else {
+            num_peaks * 4 + 1
+        };
+
+        let mut elements = Vec::with_capacity(len);
+        elements[0] = Felt::new(mmr.num_leaves as u64);
+        elements[1..].copy_from_slice(&mmr.peaks.as_slice().concat());
```

Comment:
suggestion:

- add a method `to_hash_data` to the `MmrPeaks`, that would return `Vec<Word>` with padding to be hashed
- use that on `hash_peaks` and here, here it would use similar code as this line `peaks.to_hash_data().as_slice().concat()` to get the flat version of the data.
   - then push the num_leaves on the first position

The above should make the code simpler

Reply:
Nice suggestion.  I've updated the logic to adopt this approach.